esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1504,22 +1553,16 @@ bool CWsESDLConfigEx::onGetESDLDefinition(IEspContext &context, IEspGetESDLDefin
         else
             resp.setId(id.str());
 
-        definitionTree.set(m_esdlStore->fetchDefinition(id.toLowerCase()));
+        m_esdlStore->fetchDefinitionXML(id.toLowerCase(), definition);
+        defInfo.setown(m_esdlStore->fetchDefinitionInfo(id.toLowerCase()));
 
-        if (definitionTree)
+        if (definition.length() == 0 )
         {
-            toXML(definitionTree, definition, 0,0);
-
-            if (definition.length() == 0 )
-            {
-                respcode = -1;
-                message.append("\nDefinition appears to be empty!");
-            }
-            else
-                message.setf("Successfully fetched ESDL Defintion: %s from Dali.", id.str());
+            respcode = -1;
+            message.append("\nDefinition appears to be empty!");
```

Comment:
Length of 0 is the definition of empty.

Reply:
You mean this warning message should be more assertive by stating that the definition *is* empty?

Reply:
Yes. While you're here, include the ID in the message to make debugging easier.

Reply:
Ok done.